mapie/calibration.py

```diff
@@ -0,0 +1,555 @@
+    arXiv:2107.08353 (2021).
+
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from mapie.calibration import MapieCalibrator
+    >>> X_toy = np.arange(9).reshape(-1, 1)
+    >>> y_toy = np.stack([0, 0, 1, 0, 1, 2, 1, 2, 2])
+    >>> mapie = MapieCalibrator().fit(X_toy, y_toy, random_state=20)
+    >>> y_calib = mapie.predict_proba(X_toy)
+    >>> print(y_calib)
+    [[0.84900723        nan        nan]
+     [0.75432411        nan        nan]
+     [0.62285341        nan        nan]
+     [       nan 0.33333333        nan]
+     [       nan 0.33333333        nan]
+     [       nan 0.33333333        nan]
+     [       nan        nan 0.33333002]
+     [       nan        nan 0.54326683]
+     [       nan        nan 0.66666124]]
+    """
+
+    fit_attributes = [
+        "estimator",
+        "calibrators",
+    ]
+
+    named_calibrators = {
+        "sigmoid": _SigmoidCalibration(),
+        "isotonic": IsotonicRegression(out_of_bounds="clip")
+    }
+
+    valid_methods = ["top_label"]
+
+    valid_inputs = ["multiclass", "binary"]
+
+    def __init__(
+        self,
+        estimator: Optional[ClassifierMixin] = None,
+        method: str = "top_label",
+        calibrator: Optional[Union[str, RegressorMixin]] = None,
+        cv: Optional[str] = "split",
+    ) -> None:
+        self.estimator = estimator
+        self.method = method
+        self.calibrator = calibrator
+        self.cv = cv
+
+    def _check_calibrator(
+        self,
+        calibrator: Optional[Union[str, RegressorMixin]],
+    ) -> RegressorMixin:
+        """
+        Check the input that has been provided for calibrator and
+        check that the calibrator is a valid estimator to calibrate.
+
+        Parameters
+        ----------
+        calibrator : Union[str, RegressorMixin]
+            If calibrator is a string then it returns the corresponding
+            estimator of ``named_calibrators``, else returns calibrator.
+
+            By defaults ``None``. If ``None``, set to ``sigmoid``.
+
+        Returns
+        -------
+        RegressorMixin
+            RegressorMixin calibrator.
+
+        Raises
+        ------
+        ValueError
+            If calibrator is not a key of ``named_calibrators``.
+        """
+        if calibrator is None:
+            calibrator = "sigmoid"
+
+        if isinstance(calibrator, str):
+            if calibrator in self.named_calibrators.keys():
+                calibrator = self.named_calibrators[calibrator]
+            else:
+                raise ValueError(
+                    "Please provide a string in: "
+                    + (", ").join(self.named_calibrators.keys()) + "."
+                )
+        check_estimator_fit_predict(calibrator)
+        return calibrator
+
+    def _get_labels(
+        self,
+        X: ArrayLike
+    ) -> Tuple[NDArray, NDArray]:
+        """
+        This method depends on the value of``method`` and collects the labels
```

Comment:
of``method`` -> of ``method``
miss a space

Reply:
done.